tests/test_admin_using_versioning.py

```diff
@@ -0,0 +1,164 @@
+        to be created"""
+        pagecontent = PageContentWithVersionFactory(template="page.html")
+        placeholder = PlaceholderFactory(slot="content", source=pagecontent)
+        PlaceholderFactory(slot="navigation", source=pagecontent)
+        add_plugin(placeholder, "TextPlugin", pagecontent.language, body="Test text")
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "duplicate", pagecontent.pk),
+                data={"site": Site.objects.first().pk, "slug": "foo bar"},
+                follow=True,
+            )
+        self.assertRedirects(response, self.get_admin_url(PageContent, "changelist"))
+        new_pagecontent = PageContent._base_manager.latest("pk")
+        new_placeholder = new_pagecontent.placeholders.get(slot="content")
+        self.assertEqual(PageContent._base_manager.count(), 2)
+        self.assertNotEqual(pagecontent, new_pagecontent)
+        self.assertNotEqual(pagecontent.page, new_pagecontent.page)
+        self.assertEqual(pagecontent.language, new_pagecontent.language)
+        self.assertEqual(
+            new_pagecontent.page.get_slug(new_pagecontent.language), "foo-bar"
+        )
+        new_plugins = list(downcast_plugins(new_placeholder.get_plugins_list()))
+
+        self.assertEqual(len(new_plugins), 1)
+        self.assertEqual(new_plugins[0].plugin_type, "TextPlugin")
+        self.assertEqual(new_plugins[0].body, "Test text")
+
+    def test_post_with_parent(self):
+        pagecontent1 = PageContentWithVersionFactory(
+            template="page.html",
+            page__node__depth=0,
+            page__node__path="0001",
+            page__node__numchild=1,
+        )
+        PageUrl.objects.create(
+            slug="foo",
+            path="foo",
+            language=pagecontent1.language,
+            page=pagecontent1.page,
+        )
+        pagecontent2 = PageContentWithVersionFactory(
+            template="page.html",
+            language=pagecontent1.language,
+            page__node__parent_id=pagecontent1.page.node_id,
+            page__node__depth=1,
+            page__node__path="00010001",
+        )
+        placeholder = PlaceholderFactory(slot="content", source=pagecontent2)
+        add_plugin(placeholder, "TextPlugin", pagecontent2.language, body="Test text")
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "duplicate", pagecontent2.pk),
+                data={"site": Site.objects.first().pk, "slug": "bar"},
+                follow=True,
+            )
+        self.assertRedirects(response, self.get_admin_url(PageContent, "changelist"))
+        new_pagecontent = PageContent._base_manager.latest("pk")
+        new_placeholder = new_pagecontent.placeholders.get(slot="content")
+        self.assertEqual(PageContent._base_manager.count(), 3)
```

Comment:
We should follow the AAA pattern with tests, this will need to be spaced out accordingly: https://jamescooke.info/arrange-act-assert-pattern-for-python-developers.html

Reply:
I've added carriage returns to separate it into blocks.